tests/test_opnieuw.py

```diff
@@ -65,9 +66,44 @@ def test_retry_times(self):
         except TypeError as e:
             end = time.time()
             t_diff = end - start
-            self.assertGreater(t_diff, 1)
+            self.assertGreater(t_diff, 0.5)
             self.assertEqual(self.counter, 3)
 
+    def test_retry_immediately_global(self) -> None:
+        start = time.time()
+        self.counter = 0
+        with retry_immediately():
+            try:
+                self.foo()
+            except TypeError as e:
+                end = time.time()
+                t_diff = end - start
+                self.assertLess(t_diff, 0.5)
+                self.assertEqual(self.counter, 3)
+
+    @retry(
+        retry_on_exceptions=ValueError,
+        max_calls_total=60,
+        retry_window_after_first_call_in_seconds=120,
+        namespace="bar_retry",
+    )
+    def namespaced_retry_foo(self) -> None:
+        self.counter += 1
+        raise ValueError
+
+    def test_retry_immediately(self) -> None:
+        start = time.time()
+        self.counter = 0
+        with retry_immediately("bar_retry"):
+            try:
+                self.namespaced_retry_foo()
+
+            except ValueError as e:
+                end = time.time()
+                t_diff = end - start
+                self.assertLess(t_diff, 1)
+                self.assertEqual(self.counter, 60)
+
```

Comment:
Shouldn’t we also test the mixed case — that a `with retry_immediately('foo')` does not interfere with an `@retry(namespace='bar')`? That seems like the most interesting case to me :see_no_evil:

Reply:
Awesome :)